cpu/cc2538/periph/pm.c

```diff
@@ -1,32 +1,74 @@
 /*
- * Copyright (C) 2017 Kaspar Schleiser <kaspar@schleiser.de>
- *               2017 Freie Universität Berlin
+ * Copyright (C) 2020 ML!PA Consulting GmbH
  *
  * This file is subject to the terms and conditions of the GNU Lesser
  * General Public License v2.1. See the file LICENSE in the top level
  * directory for more details.
  */
 
 /**
- * @ingroup     cpu_cortexm_common
+ * @ingroup     cpu_cc2538
  * @ingroup     drivers_periph_pm
  * @{
  *
  * @file
- * @brief       common periph/pm functions
+ * @brief       Implementation of the kernels power management interface
  *
- * @author      Kaspar Schleiser <kaspar@schleiser.de>
- * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
  *
  * @}
  */
 
-#include "cpu.h"
+#include "vendor/hw_nvic.h"
 #include "periph/pm.h"
 
-#ifdef PROVIDES_PM_SET_LOWEST_CORTEXM
-void pm_set_lowest(void)
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+void pm_set(unsigned mode)
 {
-    /* don't do anything here */
+    bool deep = 0;
+    bool switch_osc = false;
+
+    switch (mode) {
```

Comment:
would be nice to have docu for each mode, i.e., what wake up sources and features are available. Basically copy your PR description.